### ISSUE TITLE
Cannot login to application when deployed on-premise

### ISSUE THREAD

**Issue by Norbiox** [gradio-app/gradio#3171]
### Describe the bug

I cannot login to my gradio app when it's deployed in docker container on any server. I've tried on 2 different machines with different docker versions. After passing my credentials I'm not redirected to main but login page is refreshed. I've encountered this problem only on versions 3.17.x and 3.18.0 and only when deployed on some server. Everything works perfectly fine on my local machine, both when run locally or via docker.  

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

1. Dockerize gradio app with gradio authentication enabled and deploy on some server while exposing it
2. Open the app in your browser
3. Log in
4. Encounter page refresh with no redirection to actual app

### Logs

```shell
I have no logs for this. There's no error or logs neither on python side nor browser.
```

### System Info

```shell
Gradio version: 3.17.x, 3.18.0 (it worked previously on 3.16.2)
```

### Severity

blocking upgrade to latest gradio version

**Comment by Norbiox** [comment]
Hi,
Yes, I have enabled third party cookies.
I can't find this request. I'm attaching screenshots from before login and immediately after.

Before:
![before_login](https://user-images.githubusercontent.com/22893882/218649718-267f50e6-2c36-4b72-ac9b-e3d44e5a90e4.png)

After:
![after_login](https://user-images.githubusercontent.com/22893882/218649756-c0b6b859-9e7f-4646-975b-02d184e8f99d.png)

**Comment by nerochiaro** [comment]
I had the same issue and I was getting this warning in the firefox console:
`Cookie “access-token” has been rejected because a non-HTTPS cookie can’t be set as “secure”.`

Downgrading gradio to 3.16.2 allows login but obviously it is not a very good solution.
If I check the cookie set by gradio with version 3.16.2 it is indeed *not* set as "secure".

**Comment by freddyaboulton** [comment]
Thanks for getting back to me @Norbiox and @nerochiaro ! 

I'm having trouble reproducing this issue. 

I created this simple [app ](https://github.com/freddyaboulton/graido-auth) with auth (ignore the typo in the repo name 🙈). It is packaged with Docker and works well for me locally and when I deployed it to this url: https://gradio-auth.onrender.com/ . 

I tried in Chrome with and without incognito mode. And also microsoft edge and I was able to log in correctly. There's a gif of that here: 

![auth_spaces](https://user-images.githubusercontent.com/41651716/218817090-31d99084-d051-4636-aea5-1f476c4f9dc8.gif)

My guess is that for some reason, your browser is rejecting the gradio cookie set during a successful login?

**Comment by nerochiaro** [comment]
>  It is packaged with Docker and works well for me locally and when I deployed it to this url: https://gradio-auth.onrender.com/ . Credentials `admin/admin`

Can you try to serve it as a public URL like the above but using HTTP instead of HTTPS ?

**Comment by freddyaboulton** [comment]
Ah ok, I think the way we set up the cookie means it can't be set when connecting via HTTP. Will experiment to make sure we can fix while not breaking gradio auth on huggingface spaces. Thanks for your help + patience! Much appreciated.

**Comment by nerochiaro** [comment]
It's probably enough to have an optional argument to enable or disable the
"secure" flag that is set on the cookie (see the commit that I linked in my
previous comment).
Leave it on by default so that nothing will break, but document the option
to turn it off so that people who need it can use it.

**Comment by steysie** [comment]
@freddyaboulton 
Same issue. Upgraded an app from 3.16.1 to 3.18 and now authentication page seems broken. The layout is full-screen width and authentication doesn't work. Inputting the correct login-password has no effect - no warnings or errors in logging. 
<img width="1502" alt="Screenshot 2023-02-17 at 12 52 26" src="https://user-images.githubusercontent.com/32229304/219563907-b9fcb943-2bb5-4260-be6b-56987c2e0b0f.png">

I switched back to 3.16.1, but I encountered another disturbing issue.

I have two repos:
1. Gradio App
2. Flask App

I need to initialize Gradio App twice, with different arguments (data + interface strings). Both instances of the Gradio App need to send and receive requests from the Flask App.

However, it turns out that one of the instances stops sending requests (seems like it logs out from the session) - with no errors or warnings. It happens when the apps are running in the same browser. Seems like the cache is overwritten when I log in the second app.

Dockerizing doesn't help, as it's clearly an issue of browser caching.

Is anyone aware of this problem?

**Comment by nerochiaro** [comment]
> However, it turns out that one of the instances stops sending requests (seems like it logs out from the session) - with no errors or warnings. It happens when the apps are running in the same browser. Seems like the cache is overwritten when I log in the second app.

Would you mind creating a separate issue for that please @steysie, so that this bug can focus on resolving the login issue?

**Comment by freddyaboulton** [comment]
Ok, I think I have a fix for the original issue in #3223 

Can you test here: http://ec2-54-144-135-68.compute-1.amazonaws.com:7860/
